### ISSUE TITLE
More comprehensive error sourcing?

### ISSUE THREAD

**Issue by Lucretiel** [getzola/zola#1144]
https://github.com/getzola/zola/blob/af0dd5ef32ca0cbe60c4455ba8ddeb80b80c67a3/components/errors/src/lib.rs#L24-L32

Currently, this is the whole implementation of Error::source. Is there any reason we don't do something more like:

```rust
match self.source {
    Some(ref err) => Some(&**err),
    None => match self.kind {
            Tera(ref err) => Some(err),
            Io(ref err) => Some(err),
            Toml(ref err) => Some(err),
            Msg(..) => None,
            // Etc
        }
    }
}
```

**Comment by Keats** [comment]
I think we don't really care about the source of the other errors, they are not useful to the end users while the Tera ones can be

**Comment by Lucretiel** [comment]
Wouldn't that mean that even in the case of the `Tera` error it might make more sense to return `Some(tera_err)` instead of `tera_err.source()`?

**Comment by Keats** [comment]
I don't think so, but I could be wrong. Easy way to figure it out is to have an error in a template right now and see the error. Then remove the .source() and retry it to see if there's an additional error showing.

**Comment by teotwaki** [comment]
@Lucretiel is this still relevant? Will you be working on this?
